### ISSUE TITLE
Blank document comment in trait raises error different from with that in function.

### ISSUE THREAD

**Issue by da2018** [rust-lang/rust#56766]
If I use document comment that doesn't document anything in function,

```Rust
fn test(){
    println!("Hello world");
    ///
}
```

it will raise error:
```
error[E0585]: found a documentation comment that doesn't document anything.
```

But if I use this in trait, the error confuses me.

```Rust
trait User{
    fn test();
    ///
}
fn main() {
    println!("Hello world");
}
```

```
error: expected one of `async`, `const`, `extern`, `fn`, `type`, or `unsafe`, found `}`
 --> src/main.rs:4:1
  |
3 |     ///
  |        - expected one of `async`, `const`, `extern`, `fn`, `type`, or `unsafe` here
4 | }
  | ^ unexpected token
```

**Comment by da2018** [comment]
By the way, from https://doc.rust-lang.org/stable/error-index.html#E0585 , if I use blank document comment but not in the end, it doesn't raise E0585 error, so should we adjust the error message of E0585 or adjust the parser?

**Comment by da2018** [comment]
From https://doc.rust-lang.org/stable/error-index.html#E0585 , if I use blank document comment but not in the end, it doesn't raise E0585 error, so should we adjust the error message of E0585 or adjust the parser?

**Comment by estebank** [comment]
When expecting an identifier, the parser will check if the non-identifier found was a documentation comment and emit E0585:

https://github.com/rust-lang/rust/blob/7489ee9c6f92050a12a3a3097df0a7d3737d82ec/src/libsyntax/parse/parser.rs#L830-L853

An option would be to modify `eat` to check for doc comments in the same manner and complain that it isn't documenting anything (which should always be true, as by definition it should only be happening on parts of the parser that do not expect a doc comment):

https://github.com/rust-lang/rust/blob/7489ee9c6f92050a12a3a3097df0a7d3737d82ec/src/libsyntax/parse/parser.rs#L865-L871

**Comment by lcnr** [comment]
will do

**Comment by lcnr** [comment]
I don't have a lot of time right now, in case someone else wants to fix this you are free to go ahead.

**Comment by saleemjaffer** [comment]
@estebank Can I take a shot at this? I'm new to rust.

**Comment by estebank** [comment]
@saleemjaffer feel free to do so! If you look at the linked PR I believe @lcnr was in the right track.

**Comment by saleemjaffer** [comment]
@est31 I was trying to change the `eat` function.

I thought of checking if the current token is a `DocComment`,  next token is a `CloseDelim` and the previous token is not an `OpenDelim`, then raise a `E0585 `.  But PrevToken does not have an `OpenDelim`.  

So do we proceed this way or proceed in the direction of the linked PR?

**Comment by estebank** [comment]
@saleemjaffer you can potentially modify the following and `PrevTokenKind` to track `OpenDelim`

https://github.com/rust-lang/rust/blob/7e001e5c6c7c090b41416a57d4be412ed3ccd937/src/libsyntax/parse/parser.rs#L1342-L1351

but I feel that might be overkill. Why do you need ot modify the `eat` fn? The code that @lcnr wrote seems correct to me, if the `self.bump()` after identifying the bad doc comment is removed.
